public/script.js

```diff
@@ -59,213 +59,43 @@ let map;
 let projection;
 
 /** Main function run to start animation */
-async function main() { // eslint-disable-line no-unused-vars
-  const gcr = CONSTANTS.getCapabilities;
-  const requestHTTP = `${gcr.server}service=${gcr.service}&version=${gcr.version}&request=${gcr.request}`;
-  const getCapabilities = await runXMLHTTPRequest(requestHTTP);
-  const parser = new DOMParser();
-  const xmlDoc = parser.parseFromString(getCapabilities, 'text/xml');
-  // Get layer tags in GetCapabilities XML
-  const layers = xmlDoc.getElementsByTagName('Layer');
-  for (let i = 0; i < layers.length; i++) { // Loop through all layer tags
-    try {
-      // Find the first (only) extent (dates) tag
-      const datesArray = layers[i].getElementsByTagName('Extent')[0].textContent.split(',');
-      // Add the extents to the state object
-      STATE.validDates[layers[i].getElementsByTagName('Name')[0].textContent] = datesArray;
-    } catch (error) {
-      // Layer without extent tag, which means it is not relevant
-    }
-  }
+async function main() {
+  // Run GetCapabilies request to load available dates and load into STATE
+  STATE.validDates = await runGetCapabilities();
```

Comment:
I would suggest renaming this to something like `getValiddatesFromCapabilities`. This makes it clear that the response will contain valid dates, and not the entire `GetCapabilities` output.

You could have another function called `runGetCapabilities` that would be called within that function.